### ISSUE TITLE
SubstrateMLP

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEATv2#186]
Make a new class called SubstrateMLP, which is a generalized Multi-Layer Perceptron meant to exist within HyperNEAT substrates. This should be completed after #158 (prioritize this). The new class will work as follows.

There should be a constructor for initializing a new network. It should take three parameters:
List<Substrate> like what is returned by getSubstrateInformation() in HyperNEATTask,
List<Pair<String, String>> like what is returned by getSubstrateConnectivity() in HyperNEATTask.
Network, which should be a CPPN

For each Pair<String, String> in the connectivity list, find the corresponding Substrates in the Substrate list, and create a 4D double array: The first two indices are the x/y coords in the source substrate, and the next two indices are the x/y coords in the target substrate. The actual double value stored at this index is the synaptic weight of the link connecting the neurons, and is generated by the CPPN like in the HyperNEATCPPNGenotype.

Therefore, the main instance variable of the class would be of type
List<double[][][][]> 
Each of these arrays is a bunch of connection weights.
Additionally, you need a collection to hold the arrays of neurons in each layer:
List<double[][]>
These are the activation values stored in each neuron.

The class should implement the Network interface, and all methods need to be implemented accordingly.

**Comment by leg2015** [comment]
Worked some more on process method. Still needs some work and testing however

**Comment by leg2015** [comment]
MLP class should be functional. However, still needs extensive JUnit testing. However, there is a separate issue for that so I will go ahead and close this issue.
